### ISSUE TITLE
Timeout warning when no API Gateway is used

### ISSUE THREAD

**Issue by tommedema** [serverless/serverless#7498]
# Bug Report

## Description

1. What did you do?

Ran `serverless deploy` with a `timeout > 29`

1. What happened?

It warned me:

```
Function timeout setting (60) is greater than maximum allowed timeout for HTTP API endpoint (29s). This may introduce situation where endpoint times out for succesful lambda invocation.
```

Yet I do not have any HTTP endpoints setup. I am only triggering this lambda from a schedule event.

1. What should've happened?

It should not have warned me.

1. What's the content of your `serverless.yml` file?

https://gist.github.com/tommedema/3af5c66cd1cbc2d333a591c491e23c4d

Similar or dependent issues:

- #7496

**Comment by jschuur** [comment]
I got the same warning, but in my `serverless.yml` I had both an http and schedule set up for my function (even though it was only ever going to run on schedule). Once I removed the http definition from the function, the warning went away for me (serverless framework core version 1.67.0).

Curious though that in your gist, you don't have an http end point defined for your scheduled functions and are still seeing this warning. Even your top level one is set to 300 and not 60 like in your error message.

It looks like even if you have no http endpoints defined, and you have a top level timeout set up that's greater than 29 seconds, you could still get a warning.

**Comment by tommedema** [comment]
You're right. This looks like a bug

**Comment by medikoo** [comment]
This bug was fixed with https://github.com/serverless/serverless/pull/7467 and published with v1.67.0 version. Ensure to rely on latest version of Serverless.

Closing as duplicate of https://github.com/serverless/serverless/issues/7440
